man/systemd.generator.xml

```diff
@@ -85,6 +85,11 @@
     instantiate existing templates, thus extending or overriding existing definitions. Their main purpose is to convert
     configuration files that are not native unit files dynamically into native unit files.</para>
 
+    <para>Generated units may have their configurations modified by files,
+    templates, or drop-ins placed directly into /etc/systemd/system; see
```

Comment:
Drop-ins are searched also in `/usr/lib/systemd` and /run/systemd` so this comment is misleading. I am not sure why this paragraph is needed at all - generated units are exactly that - units - and all standard rules apply to them, including using overrides, drop-ins etc.

What is really missing in `systemd.generator` man page is definition of early, normal and late paths. They are actually pretty much set in stone (compiled into systemd binary) and should be documented.

Reply:
> I am not sure why this paragraph is needed at all

The best way I can explain this is, "It may be obvious to the people who programmed it, but unless you are experienced at the deep operations of systemd, it's never explicitly stated that units created by generators at ''systemctl daemon-reload' time' *can* be overridden by unit files which are also read in at that time." It's never said that overrides are applied at all, and when they are, after the generated units are created. It may "seem obvious", but I assure you to a system administrator it isn't a guess I'd be willing to make unless I'm told it works that way.

Or to put it another way, I'm really glad this *does* work that way, but just because the logical design implies it should, doesn't mean anything explicitly states that it *does*. What does it harm to do so?

Reply:
> it's never explicitly stated that units created by generators at ''systemctl daemon-reload' time' can be overridden by unit files which are also read in at that time."

Sorry? From the same man page: "may be used to override unit files in /usr, but not those in /etc". If this statement is not clear, then please, try improving it, not adding even more vague statement.

And: " will execute those binaries very early at bootup and at configuration reload time — **before unit files are loaded**". Again - if this is not sufficiently clear, please improve it.

> What does it harm to do so?

1. "units may have their configurations modified by files, templates" - that's plain wrong. Configuration is not modified by "files" unless you mean unit definition override in which case it is already described in manual page under `WRITING GENERATORS` and is sufficiently more complicated than your one liner; nor is it "modified" by templates - you create instance of (or instantiated) template in which case **template itself** (not generated unit) is searched using usual search path, **including** generated templates.
2. "or drop-ins placed directly into /etc/systemd/system" - again, this is incomplete and hence wrong.

I think that better would be to include three directories used by generators (early, normal and late) in description of unit search path in `systemd.unit(5)`. That will make it explicit that generated units are just normal units that follow standard rules.